components/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/impl/internal/IdentityMgtComponent.java

```diff
@@ -241,8 +242,9 @@ protected void registerCachingService(CarbonCachingService cachingService, Map<S
 //    }
 
     protected void unregisterCachingService(CarbonCachingService carbonCachingService) {
-
-        IdentityMgtDataHolder.getInstance().registerCacheService(null);
+        if (this.carbonCachingService == carbonCachingService) {
```

Comment:
Yes, there might be more than one OSGI services implementing same interface. We will get unRegisterXX for any one is being deregistered. We need to make sure we re removing whatever the one we are using IFF unRegisterXX comes for the same.